test/e2e/ingress/pathtype_exact.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("[Ingress] [PathType] exact", func() {
+	f := framework.NewDefaultFramework("exact")
+
+	ginkgo.BeforeEach(func() {
+		f.NewEchoDeployment()
+	})
+
+	ginkgo.It("should choose exact location for /exact", func() {
+		if !f.IsIngressV1Ready {
+			ginkgo.Skip("Kubernetes versions is not ready for Ingress V1")
+		}
+
+		host := "exact.path"
+
+		var exactPathType = networkingv1beta1.PathTypeExact
+		ing := framework.NewSingleIngress("exact", "/exact", host, f.Namespace, framework.EchoService, 80, nil)
+		ing.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].PathType = &exactPathType
+		f.EnsureIngress(ing)
+
+		ing = framework.NewSingleIngress("exact-sufix", "/exact", host, f.Namespace, framework.EchoService, 80, nil)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, host) &&
+					strings.Contains(server, "location = /exact") &&
+					strings.Contains(server, "location /exact")
+			})
+
+		body := f.HTTPTestClient().
+			GET("/exact").
+			WithHeader("Host", host).
+			Expect().
+			Status(http.StatusOK).
+			Body().
+			Raw()
+
+		assert.NotContains(ginkgo.GinkgoT(), body, "real path=/exact/sufix")
+		assert.Contains(ginkgo.GinkgoT(), body, "real path=/exact")
+
+		body = f.HTTPTestClient().
+			GET("/exact/sufix").
+			WithHeader("Host", host).
+			Expect().
+			Status(http.StatusOK).
+			Body().
+			Raw()
+
+		assert.Contains(ginkgo.GinkgoT(), body, "real path=/exact/sufix")
+	})
```

Comment:
Simply by looking at the assertions I can not tell which one is exact which one is prefix matching.